tools/Customization/DevHome.FileExplorerSourceControlIntegration/Services/RepositoryTracking.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using DevHome.Common.Helpers;
+using DevHome.Common.Services;
+using Serilog;
+using Windows.Storage;
+
+namespace DevHome.FileExplorerSourceControlIntegration.Services;
+
+public class RepositoryTracking
+{
+    public RepoStoreOptions RepoStoreOptions
+    {
+        get; set;
+    }
+
+    public enum RepositoryChange
+    {
+        Added,
+        Removed,
+    }
+
+    private readonly FileService fileService;
+
+    private readonly object trackRepoLock = new();
+
+    private Dictionary<string, string> TrackedRepositories { get; set; } = new Dictionary<string, string>();
+
+    private readonly Serilog.ILogger log = Log.ForContext("SourceContext", nameof(RepositoryTracking));
+
+    public event Windows.Foundation.TypedEventHandler<string, RepositoryChange>? RepositoryChanged;
+
+    public RepositoryTracking(string? path)
+    {
+        if (RuntimeHelper.IsMSIX)
+        {
+            RepoStoreOptions = new RepoStoreOptions
+            {
+                RepoStoreFolderPath = ApplicationData.Current.LocalFolder.Path,
+            };
+            log.Debug("Repo Store for File Explorer Integration created under ApplicationData");
+        }
+        else
+        {
+            RepoStoreOptions = new RepoStoreOptions
+            {
+                RepoStoreFolderPath = path ?? string.Empty,
+            };
+        }
+
+        fileService = new FileService();
+        RestoreTrackedRepositoriesFomJson();
+    }
+
+    public void RestoreTrackedRepositoriesFomJson()
+    {
+        lock (trackRepoLock)
+        {
+            TrackedRepositories = fileService.Read<Dictionary<string, string>>(RepoStoreOptions.RepoStoreFolderPath, RepoStoreOptions.RepoStoreFileName);
+
+            // No repositories are currently being tracked. The file will be created on first add to repository tracking.
+            log.Debug("Repo store has just been created with the first registered repository root path");

```

Comment:
Is it possible to log what the the value of the first registered repository root path?

Reply:
There may be concerns around logging potential PII in the root path itself. I will reconsider if logging the root path (with scrubbing all potential PII) would be possible and helpful in scenarios and/or anything else that we can log here. Thanks!